### PR TITLE
[plugin.video.yelo] 0.0.2

### PR DESCRIPTION
**Pull request by shycoderX**
### Description
This plugin makes it possible to watch live streams that are available on
the Belgium Telenet Yelo application.

All the plugin does is create some simple http requests to accomplish this, alongside with
widevine which handles the DRM content as the live streams are DRM protected.

For the widevine transaction the inputstream.adaptive plugin by peak3d is being used.

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by shycoderX** [comment]
Why is my PR being overlooked? Or am I too inpatient? :P

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        self.make_request("POST", "https://api.yeloplay.be/api/v1/device/verify",
+                          verify_header(Ids["deviceId"], callbackKey), None,
+                          json_verify_header(Ids["deviceId"], Ids["webId"]),
+                          False, None, self.testing)
+
+    def open_json_from_file(self, file_name):
+        path = self.kodi_wrapper.get_addon_data_path()
+
+        if not os.path.exists(path):
+            os.mkdir(path, 0775)
```
Comment (Rechi) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: This is the deprecated octal syntax, use the new one.
```suggestion
            os.mkdir(path, 0o775)
```

File: plugin.video.yelo/resources/lib/yelo/yelo.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                    return json.load(file)
+            except IOError:
+                """ Did we login? """
+                self.login()
+
+    def save_json_to_file(self, file_name, data):
+        path = self.kodi_wrapper.get_addon_data_path()
+
+        if not os.path.exists(path):
+            os.mkdir(path, 0775)
```
Comment (Rechi) [inline plugin.video.yelo/resources/lib/yelo/yelo.py]: This is the deprecated octal syntax, use the new one.
```suggestion
            os.mkdir(path, 0o775)
```

**Comment by shycoderX** [comment]
> Please squash your 2 commits into a single one. After that, from my side, is good to go.

Done.

**Pull request merged**